### PR TITLE
New release job

### PR DESCRIPTION
**Pull request by jburel**
Add entry for new job deploying the javadoc and slice2html pages

### CODE REVIEW

File: contributing/ci-release.rst [hunk 1]
```diff
@@ -92,5 +94,12 @@ OMERO
            https://github.com/openmicroscopy/ome-release.git
         #. Runs `make clean omero`
 
+    :jenkinsjob:`OMERO-DEV-release-artifacts`
+        This job deploys the javadoc and the slice2html documentation
```
Comment (mtbc) [inline contributing/ci-release.rst]: Javadoc

File: contributing/ci-release.rst [hunk 2]
```diff
@@ -92,5 +94,12 @@ OMERO
...
+
+        #. Loops through omero-{model,common,romio,renderer,blitz,gateway-java}
```
Comment (mtbc) [inline contributing/ci-release.rst]: also server?

**Comment by jburel** [comment]
@mtbc fixed

**Comment by mtbc** [comment]
Staged at https://ci.openmicroscopy.org/job/CONTRIBUTING-merge-docs/ws/src/contributing/_build/html/ci-release.html.

**Review by mtbc — APPROVED** [review]

**Pull request merged**